### PR TITLE
Check if jwt token is expired

### PR DESCRIPTION
**Pull request by KeKs0r**
This is my attempt to solve #766

---

This change is [Reviewable](https://reviewable.io/reviews/rethinkdb/horizon/768)

### CODE REVIEW

File: client/src/auth.js [hunk 1]
```diff
@@ -127,7 +127,19 @@ export class TokenStorage {
...
+    }
+    try {
+      const meta = JSON.parse(atob(token.split('.')[1]));
+      const exp = meta.exp;
+      const now = new Date().getTime() / 1000;
+      return (now < exp);
+    } catch (e){
+      return false;
+    }
+    return Boolean(token)
```
Comment (deontologician) [inline client/src/auth.js]: I don't think this line is reachable

File: client/src/auth.js [hunk 2]
```diff
@@ -127,7 +127,19 @@ export class TokenStorage {
 
   // Whether there is an auth token for the provided authType
   hasAuthToken() {
-    return Boolean(this.get())
+    const token = this.get();
+    if(!Boolean(token)){
```
Comment (deontologician) [inline client/src/auth.js]: `!` will coerce to a boolean, so you can just do `if (!token) {`

File: client/src/auth.js [hunk 3]
```diff
@@ -127,7 +127,19 @@ export class TokenStorage {
...
+      return false;
+    }
+    try {
+      const meta = JSON.parse(atob(token.split('.')[1]));
```
Comment (deontologician) [inline client/src/auth.js]: If you run eslint it will call out a few formatting errors like extraneous semicolons

**Comment by deontologician** [comment]
Logic looks good @KeKs0r, just a few minor comments to address. Thanks for the PR!

**Comment by KeKs0r** [comment]
@deontologician Have implemented your comments.

**Comment by deontologician** [comment]
Awesome, thanks a lot :)

**Pull request merged**